packages/dev/core/src/Materials/clipPlaneMaterialHelper.ts

```diff
@@ -0,0 +1,83 @@
+import type { Effect } from "./effect";
+import type { IClipPlanesHolder } from "../Misc/interfaces/iClipPlanesHolder";
+import type { Nullable } from "../types";
+import type { Plane } from "../Maths/math.plane";
+
+/** @internal */
+export function addClipPlaneUniforms(uniforms: string[]): void {
+    if (uniforms.indexOf("vClipPlane") === -1) {
+        uniforms.push("vClipPlane");
+    }
+    if (uniforms.indexOf("vClipPlane2") === -1) {
+        uniforms.push("vClipPlane2");
+    }
+    if (uniforms.indexOf("vClipPlane3") === -1) {
+        uniforms.push("vClipPlane3");
+    }
+    if (uniforms.indexOf("vClipPlane4") === -1) {
+        uniforms.push("vClipPlane4");
+    }
+    if (uniforms.indexOf("vClipPlane5") === -1) {
+        uniforms.push("vClipPlane5");
+    }
+    if (uniforms.indexOf("vClipPlane6") === -1) {
+        uniforms.push("vClipPlane6");
+    }
+}
+
+/** @internal */
+export function prepareDefinesForClipPlanes(primaryHolder: IClipPlanesHolder, secondaryHolder: IClipPlanesHolder, defines: string[] | Record<string, any>): boolean {
+    let changed = false;
+    let clipPlane = primaryHolder.clipPlane ?? secondaryHolder.clipPlane;
+    addDefine(clipPlane, defines, "CLIPPLANE", changed);

```

Comment:
it has been factored out to do the check only once as we can not use them anyway if not defined and I do not think there is a need to check them all ?

Reply:
I think there was a need when using a custom shader material, that's why the `useClipPlane=true` option does exist but I don't remember in which cases it could be useful...

[...] Actually it's the other way around, there's a need to disable all clip planes even if defined at the scene level:

https://forum.babylonjs.com/t/slice-doesnt-hide-the-edges/14910/8

So, it seems that checking for `useClipPlane==false` is enough indeed.